src/services/accounts/AccountsAssetsService.ts

```diff
@@ -0,0 +1,115 @@
+import { ApiPromise } from '@polkadot/api';
+import { StorageKey } from '@polkadot/types';
+import { AssetId, BlockHash } from '@polkadot/types/interfaces';
+import { AccountId } from '@polkadot/types/interfaces/runtime';
+import { AnyNumber } from '@polkadot/types/types';
+
+import { IAccountAssetApproval } from '../../types/responses';
+import { IAccountAssetsBalanceVec } from '../../types/responses';
+import { AbstractService } from '../AbstractService';
+
+export class AccountsAssetsService extends AbstractService {
+	constructor(api: ApiPromise) {
+		super(api);
+	}
+
+	async fetchAssetBalances(
+		hash: BlockHash,
+		address: string,
+		assets: number[] | []
+	): Promise<IAccountAssetsBalanceVec> {
+		const { api } = this;
+
+		const { number } = await api.rpc.chain.getHeader(hash);
+
+		let response;
+		if (assets.length === 0) {
+			/**
+			 * This will query all assets and return them in an array
+			 */
+			const keys = await api.query.assets.account.keysAt(hash);
+			const assetIds = this.extractAssetIds(keys);
+
+			const queryAllAssets = async () =>
```

Comment:
Should this be a helper function. Looks like its used twice in very similar ways.

Reply:
Yeah looks like both branches of the if statement have the same logic, just differ in mapping over assetIds vs assets.

Reply:
Fixed!